recipe-server/client/control/components/RecipeForm.js

```diff
@@ -210,16 +221,32 @@ export const formConfig = {
  */
 export function initialValuesWrapper(Component) {
   function Wrapped(props) {
-    const { recipe, location } = props;
+    const { recipe, location, selectedAction } = props;
     let initialValues = recipe;
     if (location.state && location.state.selectedRevision) {
       initialValues = location.state.selectedRevision;
     }
+
+    // If we still don't have initial values, roll with the defaults.
+    if (!initialValues) {
+      initialValues = { ...DEFAULT_FORM_VALUES };
+
+      // ActionField subclasses define their own initial values.
+      if (selectedAction) {
+        const ActionFields = RecipeForm.argumentsFields[selectedAction];
+        initialValues.arguments = { ...ActionFields.initialValues };
+      } else {
+        // DEFAULT_FORM_VALUES wouldn't copy this, so we add it here instead.
+        initialValues.arguments = {};
```

Comment:
Instead of `else`ing this, we could probably just do:

```
initialValues = { ...DEFAULT_FORM_VALUES, arguments: {} };
if (selectedAction) { ... }
```